api/objects.proto

```diff
@@ -239,6 +239,10 @@ message Secret {
 	// Name of the secret, so that we can use it easily for lookups.  This is
 	// just a copy of the data in `SecretValue.Spec.Annotations.Name`.
 	string name = 4;
+
+	// Whether the secret is an internal secret (not set by a user) or not.
+	// This is just a copy of the data in `SecretValue.Spec.Internal`
```

Comment:
`SecretValue.Spec` shouldn't have an internal field. A user shouldn't be able to decide wether a secret is internal or not.